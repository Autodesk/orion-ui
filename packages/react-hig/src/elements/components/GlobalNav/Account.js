/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/

import * as PropTypes from 'prop-types';
import HIGElement from '../../HIGElement';
import createComponent from '../../../adapters/createComponent';

export class Account extends HIGElement {
  commitUpdate(updatePayload, oldProps, newProp) {
    const mapping = {
      image: 'setImage',
      label: 'setLabel'
    };

    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case mapping[propKey]:
          this.hig[mapping[propKey]](propValue);
          break;
        case 'active':
          if (propValue) {
            this.hig.activate();
          } else {
            this.hig.deactivate();
          }
          break;          
        default:
          this.commitPropChange(propKey, propValue);
      }
    }
  }

  componentDidMount(){
    if (this.initialProps.activatedAccount == true ) {
      
    }
  }
}

const AccountComponent = createComponent(Account);

AccountComponent.propTypes = {
  image: PropTypes.string,
  label: PropTypes.string,
  activate: PropTypes.func,
  deactivate: PropTypes.func,
  onClick: PropTypes.func
};

AccountComponent.__docgenInfo = {
  props: {
    label: {
      description: 'sets {String} the label displayed for an account in Account/account switcher'
    },
    image: {
      description: 'sets {String} the image displayed for an account in Account/account switcher'
    },
    activate: {
      description: '{func} activates the Account'
    },
    deactivate: {
      description: '{func} deactivates the Account'
    },
    onclick: {
      description: '{func} calls the provided callback when user clicks on the Account'
    }
  }
};

export default AccountComponent;
