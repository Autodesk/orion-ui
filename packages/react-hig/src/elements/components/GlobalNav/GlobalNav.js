/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import * as HIG from 'hig.web';
import * as PropTypes from 'prop-types';

import HIGElement from '../../HIGElement';
import HIGChildValidator from '../../HIGChildValidator';
import createComponent from '../../../adapters/createComponent';

import SideNavComponent, { SideNav } from './SideNav';
import TopNavComponent, { TopNav } from './TopNav';
import SubNavComponent, { SubNav } from './SubNav';
import Slot from './Slot';

class GlobalNav extends HIGElement {
  constructor(initialProps) {
    super(HIG.GlobalNav, initialProps);
  }

  componentDidMount() {
    // Add any children
    if (this.sideNav) {
      this.hig.addSideNav(this.sideNav.hig);
      this.sideNav.componentDidMount();
    }

    if (this.topNav) {
      this.hig.addTopNav(this.topNav.hig);
      this.topNav.componentDidMount();
    }

    if (this.subNav) {
      this.hig.addSubNav(this.subNav.hig);
      this.subNav.componentDidMount();
    }

    if (this.initialProps.sideNavOpen) {
      this.hig.showSideNav();
    } else {
      this.hig.hideSideNav();
    }

    if (this.slot) {
      this.hig.addSlot(this.slot);
    }
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case TopNav:
        return new TopNav(this.hig.partials.TopNav, props);
      case SubNav:
        return new SubNav(this.hig.partials.SubNav, props);
      case SideNav:
        return new SideNav(this.hig.partials.SideNav, props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  appendChild(instance, beforeChild = {}) {
    if (instance instanceof SideNav) {
      if (this.sideNav) {
        throw new Error('only one SideNav is allowed');
      } else {
        this.sideNav = instance;
        if (this.mounted) {
          this.hig.addSideNav(instance.hig);
          instance.componentDidMount();
        }
      }
    } else if (instance instanceof TopNav) {
      if (this.topNav) {
        throw new Error('only one TopNav is allowed');
      } else {
        this.topNav = instance;
        if (this.mounted) {
          this.hig.addTopNav(instance.hig);
          instance.componentDidMount();
        }
      }
    } else if (instance instanceof SubNav) {
      if (this.SubNav) {
        throw new Error('only one SubNav is allowed');
      } else {
        this.subNav = instance;
        if (this.mounted) {
          this.hig.addSubNav(instance.hig);
          instance.componentDidMount();
        }
      }
    } else {
      throw new Error('unknown type');
    }
  }

  addSlot(element) {
    if (this.mounted) {
      this.hig.addSlot(element);
    } else {
      this.slot = element;
    }
  }

  insertBefore(instance) {
    this.appendChild(instance);
  }

  removeChild(instance) {
    if (instance instanceof SideNav) {
      this.sideNav = null;
    } else if (instance instanceof TopNav) {
      this.topNav = null;
    } else if (instance instanceof SubNav) {
      this.subNav = null;
    }
    instance.unmount();
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case 'sideNavOpen': {
          if (propValue) {
            this.hig.showSideNav();
          } else {
            this.hig.hideSideNav();
          }
          break;
        }
        case 'children': {
          /* no-op */
          break;
        }
        default: {
          console.warn(`${propKey} is unknown`);
        }
      }
    }
  }
}

const GlobalNavComponent = createComponent(GlobalNav);

GlobalNavComponent.propTypes = {
  sideNavOpen: PropTypes.bool,
  children: HIGChildValidator([
    SideNavComponent,
    TopNavComponent,
    SubNavComponent,
    Slot
  ])
};

GlobalNavComponent.__docgenInfo = {
  props: {
    sideNavOpen: {
      description: 'show or hide the SideNav'
    },

    children: {
      description: 'support adding SideNav, SubNav, or TopNav'
    }
  }
};

GlobalNavComponent.SideNav = SideNavComponent;
GlobalNavComponent.TopNav = TopNavComponent;
GlobalNavComponent.SubNav = SubNavComponent;
GlobalNavComponent.Slot = Slot;

export default GlobalNavComponent;
