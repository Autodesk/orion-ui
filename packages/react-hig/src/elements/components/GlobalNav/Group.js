/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import createComponent from '../../../adapters/createComponent';
import HIGElement from '../../HIGElement';
import HIGNodeList from '../../HIGNodeList';
import HIGChildValidator from '../../HIGChildValidator';
import ModuleComponent, { Module } from './Module';

export class Group extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);
    this.modules = new HIGNodeList({
      type: Module,
      HIGConstructor: this.hig.partials.Module,
      onAdd: (instance, beforeInstance) => {
        this.hig.addModule(instance, beforeInstance);
      }
    });

    this.state = {};
    this._render = this._render.bind(this);
  }

  componentDidMount() {
    this.modules.componentDidMount();
    this._render();
  }

  createElement(ElementConstructor, props) {
    return this.modules.createElement(ElementConstructor, props);
  }

  insertBefore(instance, insertBeforeIndex) {
    this.modules.insertBefore(instance, insertBeforeIndex);
    if (this.mounted) {
      this._render();
    }
  }

  removeChild(instance) {
    this.modules.removeChild(instance);
  }

  commitUpdate(updatePayload, oldProps, newProps) {
    this.processUpdateProps(updatePayload).then(this._render);
  }

  isVisible() {
    return this.state.isVisible;
  }

  expandModules() {
    this.modules.forEach(module => {
      module.commitUpdate({query: undefined})
      module.expandAll();
    });
    this._render();
  }

  collapseModules() {
    this.modules.forEach(module => {
      this.commitUpdate({query: undefined})
      module.collapseAll();
    });
    this._render();
  }

  _render() {
    const matches = this.modules.map(module => {
      module.commitUpdate({
        query: this.props.query,
        activeModule: this.props.activeModule,
        activeSubmodule: this.props.activeSubmodule
      });

      return module.isVisible();
    });

    if (matches.some(m => m)) {
      this.hig.show();
      this.state.isVisible = true;
    } else {
      this.hig.hide();
      this.state.isVisible = false;
    }
  }
}

const GroupComponent = createComponent(Group);

GroupComponent.propTypes = {
  children: HIGChildValidator([ModuleComponent])
};

GroupComponent.__docgenInfo = {
  props: {
    children: {
      description: 'support adding Module'
    }
  }
};

GroupComponent.Module = ModuleComponent;

export default GroupComponent;
