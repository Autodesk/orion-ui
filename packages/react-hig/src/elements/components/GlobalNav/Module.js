/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import * as PropTypes from 'prop-types';
import createComponent from '../../../adapters/createComponent';
import HIGElement from '../../HIGElement';
import HIGNodeList from '../../HIGNodeList';
import HIGChildValidator from '../../HIGChildValidator';
import SubmoduleComponent, { Submodule } from './Submodule';

export class Module extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);
    this.submodules = new HIGNodeList({
      type: Submodule,
      HIGConstructor: this.hig.partials.Submodule,
      onAdd: (instance, beforeInstance) => {
        this.hig.addSubmodule(instance, beforeInstance);
      }
    });
    this.state = {
      title: initialProps.title
    };
  }

  commitUpdate(updatePayload, oldProps, newProps) {
    const mapping = {
      icon: 'setIcon',
      title: 'setTitle',
      link: 'setLink'
    };

    this.commitUpdateWithMapping(updatePayload, mapping);

    if (Object.keys(updatePayload).includes('title')) {
      this.state.title = updatePayload.title;
    }
  }

  componentDidMount() {
    this.submodules.componentDidMount();
  }

  createElement(ElementConstructor, props) {
    return this.submodules.createElement(ElementConstructor, props);
  }

  insertBefore(instance, insertBeforeIndex) {
    this.submodules.insertBefore(instance, insertBeforeIndex);
  }

  removeChild(instance) {
    this.submodules.removeChild(instance);
  }

  matches(query) {
    return this.state.title.toLowerCase().match(query.toLowerCase());
  }

  filter(query) {
    const childMatches = this.submodules.map(submodule => {
      return submodule.filter(query);
    });

    if (childMatches.some(m => m) || this.matches(query)) {
      this.hig.show();
      return true;
    } else {
      this.hig.hide();
      return false;
    }
  }
}

const ModuleComponent = createComponent(Module);

ModuleComponent.propTypes = {
  icon: PropTypes.string,
  title: PropTypes.string,
  link: PropTypes.string,
  show: PropTypes.func,
  hide: PropTypes.func,
  activate: PropTypes.func,
  deactivate: PropTypes.func,
  showSubmodules: PropTypes.func,
  hideSubmodules: PropTypes.func,
  submodulesClosed: PropTypes.bool,
  addSubmodule: PropTypes.func,
  onClick: PropTypes.func,
  onHover: PropTypes.func,
  children: HIGChildValidator([SubmoduleComponent])
};

ModuleComponent.__docgenInfo = {
  props: {
    icon: {
      description: 'sets the icon of an Module'
    },

    title: {
      description: 'sets the title of an Module'
    },

    link: {
      description: 'sets the link of an Module'
    },

    show: {
      description: 'show (used for filtering)'
    },

    hide: {
      description: 'hide (used for filtering)'
    },

    activate: {
      description: 'activates the module'
    },

    deactivate: {
      description: 'deactivates the module'
    },

    showSubmodules: {
      description: 'show the submodules'
    },

    hideSubmodules: {
      description: 'hide the submodules'
    },

    addSubmodule: {
      description: 'add submodule'
    },

    onClick: {
      description: 'triggered when a link is clicked on'
    },

    onHover: {
      description: 'triggered when a link is hovered over'
    },

    children: {
      description: 'support adding SubModule'
    }
  }
};

ModuleComponent.Submodule = SubmoduleComponent;

export default ModuleComponent;
