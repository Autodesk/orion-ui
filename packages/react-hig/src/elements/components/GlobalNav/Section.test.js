/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import { mount } from 'enzyme';
import * as HIG from 'hig.web';
import React from 'react';

import GlobalNav from './GlobalNav';
import Section from './Section';
import Group from './Group';
import Module from './Module';
import Submodule from './Submodule';
import Collapse from './Collapse'

const SubmoduleContext = props => {;
  return (
    <GlobalNav>
      <GlobalNav.SideNav>
        <GlobalNav.SideNav.SectionList>
          <Section headerLabel='Project' headerName="ThunderStorm" >
             <Collapse isCollapsed={props.isCollapsed} />
             <Group>
              <Module icon="gear" title="Contractor" submodulesClosed={props.isCollapsed} >
                <Submodule title="Library" link='#' />  
              </Module>   
            </Group>  
            
          </Section>
        </GlobalNav.SideNav.SectionList>
      </GlobalNav.SideNav>
    </GlobalNav>
  );
};

const Context = props => {
  const { children, ...rest } = props;
  return (
    <GlobalNav>
      <GlobalNav.SideNav>
        <GlobalNav.SideNav.SectionList>
          <Section {...rest}>
            {children}
          </Section>
        </GlobalNav.SideNav.SectionList>
      </GlobalNav.SideNav>
    </GlobalNav>
  );
};

function higContext(defaults) {
  const higContainer = document.createElement('div');

  // use spread here to clone defaults since HIG.Button mutates this object
  const higNav = new HIG.GlobalNav();

  higNav.mount(higContainer);

  const higSideNav = new higNav.partials.SideNav();
  higNav.addSideNav(higSideNav);

  const higSection = new higSideNav.partials.Section(defaults);

  higSideNav.addSection(higSection);

  return { higNav, higSideNav, higSection, higContainer };
}

describe('<Section>', () => {
  describe('headerLabel', () => {
    it('sets the headerLabel default', () => {
      const defaults = { headerLabel: 'some label' };

      const { higContainer } = higContext(defaults);

      const reactContainer = document.createElement('div');

      const wrapper = mount(<Context {...defaults} />, {
        attachTo: reactContainer
      });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      expect(reactContainer.firstChild.outerHTML).toEqual(
        higContainer.firstChild.outerHTML
      );
    });

    it('updates the headerLabel', () => {
      const defaults = { headerLabel: 'some label' };

      const { higSection, higContainer } = higContext(defaults);

      const reactContainer = document.createElement('div');

      const wrapper = mount(<Context {...defaults} />, {
        attachTo: reactContainer
      });

      const newHeaderLabel = 'new header label';

      // hig.web API
      higSection.setHeaderLabel(newHeaderLabel);

      // React API
      wrapper.setProps({ headerLabel: newHeaderLabel });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      expect(reactContainer.firstChild.outerHTML).toEqual(
        higContainer.firstChild.outerHTML
      );
    });

    it('logs an error if the headerLabel is not the correct type', () => {
      global.console.error = jest.fn();

      mount(<Context headerLabel={[]} />);

      expect(console.error).toBeCalledWith(
        expect.stringMatching(
          /Invalid prop `headerLabel` of type `array` supplied to `Section`, expected `string`./
        )
      );
    });
  });

  describe('headerName', () => {
    it('sets the headerName default', () => {
      const defaults = { headerName: 'some label' };

      const { higContainer } = higContext(defaults);

      const reactContainer = document.createElement('div');

      const wrapper = mount(<Context {...defaults} />, {
        attachTo: reactContainer
      });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      expect(reactContainer.firstChild.outerHTML).toEqual(
        higContainer.firstChild.outerHTML
      );
    });

    it('updates the headerName', () => {
      const defaults = { headerName: 'some label' };

      const { higSection, higContainer } = higContext(defaults);

      const reactContainer = document.createElement('div');

      const wrapper = mount(<Context {...defaults} />, {
        attachTo: reactContainer
      });

      const newHeaderName = 'new header name';

      // hig.web API
      higSection.setHeaderName(newHeaderName);

      // React API
      wrapper.setProps({ headerName: newHeaderName });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      expect(reactContainer.firstChild.outerHTML).toEqual(
        higContainer.firstChild.outerHTML
      );
    });

    it('logs an error if the headerName is not the correct type', () => {
      global.console.error = jest.fn();

      mount(<Context headerName={[]} />);

      expect(console.error).toBeCalledWith(
        expect.stringMatching(
          /Invalid prop `headerName` of type `array` supplied to `Section`, expected `string`./
        )
      );
    });
  });

  describe('children: <Group>', () => {
    it('can render a list of <Group> elements', () => {
      const defaults = { headerLabel: 'some label' };

      const { higSection, higContainer } = higContext(defaults);

      var group1 = new higSection.partials.Group();
      higSection.addGroup(group1);

      var group2 = new higSection.partials.Group();
      higSection.addGroup(group2);

      const reactContainer = document.createElement('div');

      const wrapper = mount(
        <Context {...defaults}>
          <Group />
          <Group />
        </Context>,
        {
          attachTo: reactContainer
        }
      );

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      expect(reactContainer.firstChild.outerHTML).toEqual(
        higContainer.firstChild.outerHTML
      );
    });

    it('can insert <Group> elements before and after another <Group>', () => {
      const defaults = { headerLabel: 'some label' };

      const { higSection, higContainer } = higContext(defaults);

      var group1 = new higSection.partials.Group();

      // DELIBERATELY DON'T ADD GROUP 1

      var group2 = new higSection.partials.Group();
      higSection.addGroup(group2);

      // ADD GROUP 1 before GROUP 2
      higSection.addGroup(group1, group2);

      class CustomComponent extends React.Component {
        constructor(props) {
          super(props);
          this.state = { showingItemBefore: false, showingItemAfter: false };
        }

        render() {
          return (
            <Context {...defaults}>
              {this.state.showingItemBefore && <Group />}
              <Group />
              {this.state.showingItemAfter && <Group />}
            </Context>
          );
        }
      }

      const reactContainer = document.createElement('div');
      const wrapper = mount(<CustomComponent />, {
        attachTo: reactContainer
      });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      wrapper.setState({ showingItemBefore: true });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      expect(reactContainer.firstChild.outerHTML).toEqual(
        higContainer.firstChild.outerHTML
      );

      wrapper.setState({ showingItemBefore: false });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      wrapper.setState({ showingItemAfter: true });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      wrapper.setState({ showingItemAfter: false });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();
    });

    it('can not render HTML elements as children', () => {
      global.console.error = jest.fn();

      mount(
        <Context>
          <div>Hello world!</div>
        </Context>
      );

      expect(console.error).toBeCalledWith(
        expect.stringMatching(
          /'div' is not a valid child of Section. Children should be of type 'Group, Collapse'/
        )
      );
    });

    it('can not render HTML text as children', () => {
      global.console.error = jest.fn();

      mount(
        <Context>
          Hello world!
        </Context>
      );

      expect(console.error).toBeCalledWith(
        expect.stringMatching(
          /'Hello world!' is not a valid child of Section. Children should be of type 'Group, Collapse'/
        )
      );
    });
  });

  describe('submodules', () => {
    it('can show submodules when submodulesCollapsed is false', () => {
      var props = {isCollapsed: false}

      const reactContainer = document.createElement('div');

      const wrapper = mount(<SubmoduleContext {...props} />, {
        attachTo: reactContainer
      });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      expect(reactContainer.getElementsByClassName('hig__global-nav__side-nav__section__collapse--collapsed').length).toEqual(1)
      expect(reactContainer.getElementsByClassName('hig__global-nav__side-nav__section__group__module__submodules--hide').length).toEqual(0)


    });


     it('can show submodules when submodulesCollapsed is true', () => {
      var props = {isCollapsed: true}

      const reactContainer = document.createElement('div');

      const wrapper = mount(<SubmoduleContext {...props} />, {
        attachTo: reactContainer
      });

      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

      expect(reactContainer.getElementsByClassName('hig__global-nav__side-nav__section__collapse--collapsed').length).toEqual(0)
      expect(reactContainer.getElementsByClassName('hig__global-nav__side-nav__section__group__module__submodules--hide').length).toEqual(1)


    });
  });
});
