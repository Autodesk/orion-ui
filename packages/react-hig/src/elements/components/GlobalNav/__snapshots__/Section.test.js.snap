// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section> children: <Group> can insert <Group> elements before and after another <Group> 1`] = `
"<div class=\\"hig__global-nav\\">
    <div class=\\"hig__global-nav__sidenav\\">
    <div class=\\"hig__global-nav__side-nav__section\\">
    <div class=\\"hig__global-nav__side-nav__section__header\\">
        <span class=\\"hig__global-nav__side-nav__section__header__label\\">some label</span>
        <span class=\\"hig__global-nav__side-nav__section__header__name\\">X</span>
    </div>
    <div class=\\"hig__global-nav__side-nav__section__group\\">
    <!--ITEM-->
</div><!--GROUP-->
</div><!--SECTION-->
    <div class=\\"hig__global-nav__sidenav__links\\">
        <!--LINK-->
    </div>
    <div class=\\"hig__global-nav__sidenav__copywright\\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<Section> children: <Group> can insert <Group> elements before and after another <Group> 2`] = `
"<div class=\\"hig__global-nav\\">
    <div class=\\"hig__global-nav__sidenav\\">
    <div class=\\"hig__global-nav__side-nav__section\\">
    <div class=\\"hig__global-nav__side-nav__section__header\\">
        <span class=\\"hig__global-nav__side-nav__section__header__label\\">some label</span>
        <span class=\\"hig__global-nav__side-nav__section__header__name\\">X</span>
    </div>
    <div class=\\"hig__global-nav__side-nav__section__group\\">
    <!--ITEM-->
</div><div class=\\"hig__global-nav__side-nav__section__group\\">
    <!--ITEM-->
</div><!--GROUP-->
</div><!--SECTION-->
    <div class=\\"hig__global-nav__sidenav__links\\">
        <!--LINK-->
    </div>
    <div class=\\"hig__global-nav__sidenav__copywright\\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<Section> children: <Group> can insert <Group> elements before and after another <Group> 3`] = `
"<div class=\\"hig__global-nav\\">
    <div class=\\"hig__global-nav__sidenav\\">
    <div class=\\"hig__global-nav__side-nav__section\\">
    <div class=\\"hig__global-nav__side-nav__section__header\\">
        <span class=\\"hig__global-nav__side-nav__section__header__label\\">some label</span>
        <span class=\\"hig__global-nav__side-nav__section__header__name\\">X</span>
    </div>
    <div class=\\"hig__global-nav__side-nav__section__group\\">
    <!--ITEM-->
</div><!--GROUP-->
</div><!--SECTION-->
    <div class=\\"hig__global-nav__sidenav__links\\">
        <!--LINK-->
    </div>
    <div class=\\"hig__global-nav__sidenav__copywright\\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<Section> children: <Group> can insert <Group> elements before and after another <Group> 4`] = `
"<div class=\\"hig__global-nav\\">
    <div class=\\"hig__global-nav__sidenav\\">
    <div class=\\"hig__global-nav__side-nav__section\\">
    <div class=\\"hig__global-nav__side-nav__section__header\\">
        <span class=\\"hig__global-nav__side-nav__section__header__label\\">some label</span>
        <span class=\\"hig__global-nav__side-nav__section__header__name\\">X</span>
    </div>
    <div class=\\"hig__global-nav__side-nav__section__group\\">
    <!--ITEM-->
</div><div class=\\"hig__global-nav__side-nav__section__group\\">
    <!--ITEM-->
</div><!--GROUP-->
</div><!--SECTION-->
    <div class=\\"hig__global-nav__sidenav__links\\">
        <!--LINK-->
    </div>
    <div class=\\"hig__global-nav__sidenav__copywright\\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<Section> children: <Group> can insert <Group> elements before and after another <Group> 5`] = `
"<div class=\\"hig__global-nav\\">
    <div class=\\"hig__global-nav__sidenav\\">
    <div class=\\"hig__global-nav__side-nav__section\\">
    <div class=\\"hig__global-nav__side-nav__section__header\\">
        <span class=\\"hig__global-nav__side-nav__section__header__label\\">some label</span>
        <span class=\\"hig__global-nav__side-nav__section__header__name\\">X</span>
    </div>
    <div class=\\"hig__global-nav__side-nav__section__group\\">
    <!--ITEM-->
</div><!--GROUP-->
</div><!--SECTION-->
    <div class=\\"hig__global-nav__sidenav__links\\">
        <!--LINK-->
    </div>
    <div class=\\"hig__global-nav__sidenav__copywright\\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<Section> children: <Group> can render a list of <Group> elements 1`] = `
"<div class=\\"hig__global-nav\\">
    <div class=\\"hig__global-nav__sidenav\\">
    <div class=\\"hig__global-nav__side-nav__section\\">
    <div class=\\"hig__global-nav__side-nav__section__header\\">
        <span class=\\"hig__global-nav__side-nav__section__header__label\\">some label</span>
        <span class=\\"hig__global-nav__side-nav__section__header__name\\">X</span>
    </div>
    <div class=\\"hig__global-nav__side-nav__section__group\\">
    <!--ITEM-->
</div><div class=\\"hig__global-nav__side-nav__section__group\\">
    <!--ITEM-->
</div><!--GROUP-->
</div><!--SECTION-->
    <div class=\\"hig__global-nav__sidenav__links\\">
        <!--LINK-->
    </div>
    <div class=\\"hig__global-nav__sidenav__copywright\\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<Section> headerLabel sets the headerLabel default 1`] = `
"<div class=\\"hig__global-nav\\">
    <div class=\\"hig__global-nav__sidenav\\">
    <div class=\\"hig__global-nav__side-nav__section\\">
    <div class=\\"hig__global-nav__side-nav__section__header\\">
        <span class=\\"hig__global-nav__side-nav__section__header__label\\">some label</span>
        <span class=\\"hig__global-nav__side-nav__section__header__name\\">X</span>
    </div>
    <!--GROUP-->
</div><!--SECTION-->
    <div class=\\"hig__global-nav__sidenav__links\\">
        <!--LINK-->
    </div>
    <div class=\\"hig__global-nav__sidenav__copywright\\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<Section> headerLabel updates the headerLabel 1`] = `
"<div class=\\"hig__global-nav\\">
    <div class=\\"hig__global-nav__sidenav\\">
    <div class=\\"hig__global-nav__side-nav__section\\">
    <div class=\\"hig__global-nav__side-nav__section__header\\">
        <span class=\\"hig__global-nav__side-nav__section__header__label\\">new header label</span>
        <span class=\\"hig__global-nav__side-nav__section__header__name\\">X</span>
    </div>
    <!--GROUP-->
</div><!--SECTION-->
    <div class=\\"hig__global-nav__sidenav__links\\">
        <!--LINK-->
    </div>
    <div class=\\"hig__global-nav__sidenav__copywright\\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<Section> headerName sets the headerName default 1`] = `
"<div class=\\"hig__global-nav\\">
    <div class=\\"hig__global-nav__sidenav\\">
    <div class=\\"hig__global-nav__side-nav__section\\">
    <div class=\\"hig__global-nav__side-nav__section__header\\">
        <span class=\\"hig__global-nav__side-nav__section__header__label\\">Project</span>
        <span class=\\"hig__global-nav__side-nav__section__header__name\\">some label</span>
    </div>
    <!--GROUP-->
</div><!--SECTION-->
    <div class=\\"hig__global-nav__sidenav__links\\">
        <!--LINK-->
    </div>
    <div class=\\"hig__global-nav__sidenav__copywright\\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<Section> headerName updates the headerName 1`] = `
"<div class=\\"hig__global-nav\\">
    <div class=\\"hig__global-nav__sidenav\\">
    <div class=\\"hig__global-nav__side-nav__section\\">
    <div class=\\"hig__global-nav__side-nav__section__header\\">
        <span class=\\"hig__global-nav__side-nav__section__header__label\\">Project</span>
        <span class=\\"hig__global-nav__side-nav__section__header__name\\">new header name</span>
    </div>
    <!--GROUP-->
</div><!--SECTION-->
    <div class=\\"hig__global-nav__sidenav__links\\">
        <!--LINK-->
    </div>
    <div class=\\"hig__global-nav__sidenav__copywright\\">© 2017 Autodesk, Inc. </div>
</div><!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;
